src/openrct2/drawing/Sprite.cpp

```diff
@@ -713,28 +776,23 @@ extern "C"
         sint32 skipX = left - x;
         sint32 skipY = top - y;
 
-        uint8 *maskSrc = imgMask->offset + (skipY * imgMask->width) + skipX;
-        uint8 *colourSrc = imgColour->offset + (skipY * imgColour->width) + skipX;
-        uint8 *dst = dpi->bits + (left - dpi->x) + ((top - dpi->y) * (dpi->width + dpi->pitch));
+        uint8 const * maskSrc   = imgMask->offset + (skipY * imgMask->width) + skipX;
+        uint8 const * colourSrc = imgColour->offset + (skipY * imgColour->width) + skipX;
+        uint8       * dst       = dpi->bits + (left - dpi->x) + ((top - dpi->y) * (dpi->width + dpi->pitch));
 
-        sint32 maskWrap = imgMask->width - width;
+        sint32 maskWrap   = imgMask->width - width;
         sint32 colourWrap = imgColour->width - width;
-        sint32 dstWrap = ((dpi->width + dpi->pitch) - width);
-        for (sint32 yy = top; yy < bottom; yy++) {
-            for (sint32 xx = left; xx < right; xx++) {
-                uint8 colour = (*colourSrc) & (*maskSrc);
-                if (colour != 0) {
-                    *dst = colour;
-                }
+        sint32 dstWrap    = ((dpi->width + dpi->pitch) - width);
 
-                maskSrc++;
-                colourSrc++;
-                dst++;
-            }
-            maskSrc += maskWrap;
-            colourSrc += colourWrap;
-            dst += dstWrap;
+#ifdef __SSE4_1__
+        if (width == 32)
+        {
```

Comment:
Sorry for being nitpicky, but either both the `if` and `else` should have braces, or neither.